src/connector.ts

```diff
@@ -265,11 +297,56 @@ export class Connector {
     };
   }
 
-  // clear up the event loop from the internal cloud sql
+  async startLocalProxy({
+    authType,
+    ipType,
+    instanceConnectionName,
+    listenOptions,
+    serverOptions,
+  }: SocketConnectionOptions): Promise<void> {
+    const {stream} = await this.getOptions({
+      authType,
+      ipType,
+      instanceConnectionName,
+    });
+
+    // Opens a local server that listens
+    // to the location defined by `listenOptions`
+    const server = createServer(serverOptions);
+    this.localProxies.add(server);
+
+    /* c8 ignore next 3 */
+    server.once('error', err => {
+      console.error(err);
+    });
+
+    // Once a connection is stablished, pipe data from the
+    // local proxy server to the secure TCP Socket and vice-versa.
+    server.once('connection', c => {
+      const s = stream();
+      this.sockets.add(s);
+      this.sockets.add(c);
+      c.pipe(s);
+      s.pipe(c);
+    });
+
+    const listen = promisify(server.listen) as Function;
+    await listen.call(server, listenOptions);
+  }
+
+  // Clear up the event loop from the internal cloud sql
   // instances timeout callbacks that refresh instance info
+  // along with any sockets and tunnel servers that may
+  // have been used to expose a local proxy tunnel.
```

Comment:
this reads a lil funny, might want to see if we can make it more clear what is going on.